docs/samples/python/transformers_recognizer/transformer_recognizer.py

```diff
@@ -0,0 +1,289 @@
+        :return: List of the supported entities.
+        """
+        return self.supported_entities
+
+    # Class to use transformers with Presidio as an external recognizer.
+    def analyze(
+        self, text: str, entities: List[str], nlp_artifacts: NlpArtifacts = None
+    ) -> List[RecognizerResult]:
+        """
+        Analyze text using transformers model to produce NER tagging.
+        :param text(str) : The text for analysis.
+        :param entities(List[str]): Not working properly for this recognizer.
+        :param nlp_artifacts: Not used by this recognizer.
+        :return: The list of Presidio RecognizerResult constructed from the recognized
+            transformers detections.
+        """
+
+        results = list()
+        # Run transformer model on the provided text
+        ner_results = self._get_ner_results_for_text(text)
+
+        for res in ner_results:
+            res['entity_group'] = self.__check_label_transformer(
+                res["entity_group"])
+            textual_explanation = self.default_explanation.format(
+                res["entity_group"]
+            )
+            explanation = self.build_transformers_explanation(
+                round(res["score"], 2), textual_explanation, res["word"]
+            )
+            transformers_result = self._convert_to_recognizer_result(
+                res, explanation
+            )
+
+            results.append(transformers_result)
+
+        return results
+
+    @ staticmethod
+    def split_text_to_word_chunks(input_length: int, chunk_length: int, overlap_length: int) -> List[List]:
+        """The function calculates chunks of text with size chunk_length. Each chunk has overlap_length number of
+        words to create context and continuity for the model
+
+        Args:
+            input_length (int): Length of input_ids for a given text
+            chunk_length (int): Length of each chunk of input_ids. Should match the max input length of the transformer model
+            overlap_length (int): Number of overlapping words in each chunk
+
+        Returns:
+            List[List]: List of start and end position for each text chunk
+        """
+        return [[i, min([i+chunk_length, input_length])] for i in range(0, input_length, chunk_length-overlap_length)]
+
+    def _get_ner_results_for_text(self, text: str) -> List[dict]:
+        """The function runs model inference on the provided text.
+        If length of text > max_length tokens, the text is split into chunks with n = 40 overlapping characters
+        The results are aggregated and duplicates are removed.
+
+        Args:
+            text (str): The text to run inference on
+
+        Returns:
+            List[dict]: List of NER predictions on the word level
+        """
+
+        model_max_length = self.pipeline.tokenizer.model_max_length
+        # calculate inputs based on the text
+        text_length = len(text)
+        if text_length <= model_max_length*2:
```

Comment:
From the examples I looked at the average input_id  count was around 320, but I will modify the code so that all texts go trough splitting just to cover all edge cases.